app/models/ability.rb

```diff
@@ -3,13 +3,45 @@
 class Ability
   include Hydra::Ability
   include Hyrax::Ability
+  include GroupAwareRoleChecker
+  # OVERRIDE: Added custom user ability roles
+  include Hyrax::Ability::UserAbility
+  include Hyrax::Ability::WorkAbility
 
   self.ability_logic += %i[
-    everyone_can_create_curation_concerns
     group_permissions
     superadmin_permissions
+    collection_roles
+    user_roles
+    work_roles
     featured_collection_abilities
   ]
+  # If the Groups with Roles feature is disabled, allow registered users to create curation concerns
+  # (Works, Collections, and FileSets). Otherwise, omit this ability logic as to not
+  # conflict with the roles that explicitly grant creation permissions.
+  unless ENV['SETTINGS__RESTRICT_CREATE_AND_DESTROY_PERMISSIONS'] == 'true'
```

Comment:
SETTINGS__ is removed from Hyku. See comment above. Also avoid == 'true'

```suggestion
  unless ActiveModel::Type::Boolean.new.cast(
    ENV.fetch('HYKU_RESTRICT_CREATE_AND_DESTROY_PERMISSIONS', nil)
  )
```